### DISCUSSION TITLE
question: how to pass X_Forwarded_For to subgraph and get it from there

### DISCUSSION THREAD

**Discussion by billnbell2** [apollographql/router#4288] [General]
How do we pass AWS X_Forwarded_For from Cloudfront -> ALB -> Router -> ALB -> Subgraph

https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html#x-forwarded-for

We want to get the IP address so we can pass it to Algolia.

**Comment by garypen** [comment]
By default, the router won't propagate headers unless they are configured for propagation. So, you could [add config](https://www.apollographql.com/docs/router/configuration/header-propagation/#propagate) to cause the header to propagate.

However, this wouldn't achieve what you are looking for, since the router won't attach it's own IP address to the chain. I think this is an issue and we should modify the propagation logic so that, if `X-Forwarded-For` is configured for propagation, then the router behaves as expected and adds its IP address to the comma separated list.

What do you think of the above @billnbell2 ? If you like the sound of that I'll file an issue and we can proceed from there.
